audio-worklet/src/noisegate-demo.js

```diff
@@ -0,0 +1,284 @@
+/**
+ * Copyright 2017 Google Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+class NoisegateDemo {
+  /**
+   * @class NoisegateDemo
+   * @constructor Initializes and modifies noisegate settings in response to
+   * GUI input.
+   * @param {AudioContext} context The Audio Context
+   * @param {Boolean} workletIsAvailable Switch depending on browser
+   *                  which determines if both script processor and audio
+   *                  worklet should be used. If false, only the script
+   *                  processor will be used.
+   */
+  constructor(context, workletIsAvailable) {
+    this.context_ = context;
+    this.masterGain_ = new GainNode(this.context_, {gain: 0.5});
+    this.workletIsAvailable_ = workletIsAvailable;
+
+    this.attack_ = 0;
+    this.attackMax_ = 1;
```

Comment:
For the old demo (the one you heard on 08-04-2017), the audio sounded slightly better at around 250 ms. But for the new demo, higher attack and release don't buy you better sound since I make an effort in the recording to only pause in between sentences. So I will bring this back down to 0.1.